phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateSchemaIT.java

```diff
@@ -95,9 +96,33 @@ public void testCreateSchema() throws Exception {
             conn.createStatement().execute("CREATE SCHEMA \""
                     + SchemaUtil.HBASE_NAMESPACE.toUpperCase() + "\"");
 
-            assertNotNull(admin.getNamespaceDescriptor(SchemaUtil.SCHEMA_FOR_DEFAULT_NAMESPACE.toUpperCase()));
-            assertNotNull(admin.getNamespaceDescriptor(SchemaUtil.HBASE_NAMESPACE.toUpperCase()));
+            assertTrue(isHbaseNamespaceAvailable(admin, SchemaUtil.SCHEMA_FOR_DEFAULT_NAMESPACE.toUpperCase()));
+            assertTrue(isHbaseNamespaceAvailable(admin, SchemaUtil.HBASE_NAMESPACE.toUpperCase()));
 
         }
     }
+
+    /**
+     * Returns true if HBase namespace exists, else returns false
+     * @param hbaseAdmin HbaseAdmin Object
+     * @param schemaName Phoenix schema name for which we check existence of the HBase namespace
+     * @return true if the HBase namespace exists, else returns false
+     * @throws SQLException If there is an exception checking the HBase namespace
+     */
+    boolean isHbaseNamespaceAvailable(HBaseAdmin hbaseAdmin, String schemaName) throws IOException{
```

Comment:
HBaseAdmin is now listed as InterfaceAudience.Private, since it's a concrete class. If possible we should depend on the Admin interface it implements.

Reply:
Sure. I did for couple of usages but I saw we are using HbaseAdmin at multiple places in CQSI and other classes. I will file another JIRA to use Admin rather than HBaseAdmin directly.

Also, one more thing that I came across is HConnection is also deprecated and Connection should be used in it's place. (https://hbase.apache.org/1.2/apidocs/org/apache/hadoop/hbase/client/HConnection.html)

Should I go-ahead and file a JIRA for that refactoring as well ?